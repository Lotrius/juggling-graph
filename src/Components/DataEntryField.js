import React, { Component } from 'react';

class DataEntryField extends Component {
    render() {
        return (
            // Form
            <form onSubmit={this.submitData}>
                <div className='cf'>
                    
                    {/* Input field */}
                    <div>
                        <input
                            className='f6 f5-l input-reset fl black-80 bg-white pa3 lh-solid w-100 w-75-m w-80-l br2-ns br--left-ns b--black'
                            type='text'
                            placeholder='Number of catches'
                            ref="input"
                        />
                    </div>

                    {/* Submit button */}
                    <div>
                        <input
                            className='f6 f5-l button-reset fl pv3 tc bn bg-green hover-bg-green white pointer w-100 w-25-m w-20-l br2-ns br--right-ns'
                            type='submit'
                            value='Add'
                        />
                    </div>
                </div>
            </form>
        );
    }

    submitData = (event) => {
        event.preventDefault(); // Prevent page from refreshing when submitted

        const num = parseInt(this.refs.input.value, 10); // Get value
        
        // If value is actually a number update daily data and clear text field
        // Otherwise just clear text field and nothing else
        if (isNaN(num)) {
            this.refs.input.value = '';
        } else {
            this.refs.input.value = '';
            this.props.updateDailyData(num);
        }
    }
}

export default DataEntryField;