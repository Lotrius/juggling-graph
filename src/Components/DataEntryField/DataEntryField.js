import React, { Component } from 'react';
import './Input.css';
import PropTypes from 'prop-types';

// Get current login state
const guest = JSON.parse(localStorage.getItem('guest'));
const sandbox = JSON.parse(localStorage.getItem('sandbox'));

class DataEntryField extends Component {
  constructor() {
    super();
    this.state = {
      loginState: ''
    };
  }

  /**
   * Update login state once components mount
   */
  componentDidMount() {
    // Set login state based on how user logged in
    if (guest) {
      console.log('sup');
      this.setState({ loginState: '(disabled as guest)' });
    } else if (sandbox) {
      this.setState({ loginState: '(sandbox)' });
    }

    // UPDATEEE
    this.forceUpdate();
  }

  /**
   * When add button is pressed, submit data
   *
   * @param {Object} event
   */
  submitData = (event) => {
    const { updateDailyData } = this.props;

    event.preventDefault(); // Prevent page from refreshing when submitted

    const num = parseInt(event.target[0].value, 10); // Get value
    event.target.reset(); // Reset field

    updateDailyData(num);
  };

  /**
   * Render entry field
   */
  render() {
    const { loginState } = this.state;

    return (
      // Form
      <form onSubmit={this.submitData}>
        <div>
          {/* Input field */}
          <div>
            <input
              className={`f6 mb2 input-reset fl black-80 pa3 lh-solid w-100 w-75-m w-80-l br3 b--black ${
                guest ? 'bg-moon-gray' : 'bg-white'
              }`}
              style={{ outline: 'none' }}
              type="number"
              placeholder={`Number of catches ${loginState}`}
              disabled={guest || null}
            />
          </div>

          {/* Submit button */}
          <div>
            <input
              className={`f6 button-reset pv3 tc bn bg-green hover-bg-green white w-50 w-25-m w-20-l br3 ${
                guest ? null : 'pointer'
              }`}
              style={{ outline: 'none' }}
              type="submit"
              value="Add"
              disabled={guest || null}
            />
          </div>
        </div>
      </form>
    );
  }
}

DataEntryField.propTypes = {
  updateDailyData: PropTypes.func
};

DataEntryField.defaultProps = {
  updateDailyData: null
};

export default DataEntryField;
