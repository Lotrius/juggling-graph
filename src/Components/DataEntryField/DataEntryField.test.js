import { shallow } from 'enzyme';
import React from 'react';
import DataEntryField from './DataEntryField';

let wrapper;

beforeEach(() => {
  const mockStartData = [{ x: 0, y: 0 }];
  const mockUpdateDailyData = jest.fn((catches) =>
    mockStartData.push({ x: mockStartData.length, y: catches })
  );

  wrapper = shallow(<DataEntryField updateDailyData={mockUpdateDailyData} />);
});

// Check nothing changed within the file
it('Render DataEntryField component', () => {
  expect(wrapper).toMatchSnapshot();
});
