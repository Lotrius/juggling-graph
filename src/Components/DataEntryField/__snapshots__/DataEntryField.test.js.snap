// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DataEntryField component 1`] = `ShallowWrapper {}`;
